source/extensions/filters/http/oauth2/filter.h

```diff
@@ -51,23 +51,23 @@ class SDSSecretReader : public SecretReader {
                   Secret::GenericSecretConfigProviderSharedPtr token_secret_provider, Api::Api& api)
       : api_(api), client_secret_provider_(std::move(client_secret_provider)),
         token_secret_provider_(std::move(token_secret_provider)) {
-    readAndWatchSecret(client_secret_, *client_secret_provider_);
-    readAndWatchSecret(token_secret_, *token_secret_provider_);
+    readAndWatchSecret(client_secret_, *client_secret_provider_, update_callback_client_);
+    readAndWatchSecret(token_secret_, *token_secret_provider_, update_callback_token_);
   }
 
   const std::string& clientSecret() const override { return client_secret_; }
 
   const std::string& tokenSecret() const override { return token_secret_; }
 
 private:
-  void readAndWatchSecret(std::string& value,
-                          Secret::GenericSecretConfigProvider& secret_provider) {
+  void readAndWatchSecret(std::string& value, Secret::GenericSecretConfigProvider& secret_provider,
```

Comment:
Could you return the update_callback from the method instead of using a parameter? I think it would be more readable...

Reply:
Good point.